### PR TITLE
fix(cache): global cache init

### PR DESCRIPTION
**Pull request by rarkins**
Refactors the global file cache so that it is inactive unless initialized.

### CODE REVIEW

File: lib/util/cache/global/file.spec.ts [hunk 1]
```diff
@@ -1,12 +1,16 @@
 import os from 'os';
 import { init } from './file';
 
-describe('lib/workers/global/cache', () => {
+describe('lib/util/cache/global/file', () => {
```
Comment (viceice) [inline lib/util/cache/global/file.spec.ts:4]: We should use `getName(__fileName)` here

File: lib/util/cache/global/index.ts [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+import * as fileCache from './file';
+
+let initComplete = false;
 
 function getGlobalKey(namespace: string, key: string): string {
   return `global%%${namespace}%%${key}`;
 }
 
 export function get<T = any>(namespace: string, key: string): Promise<T> {
+  if (!initComplete) {
```
Comment (viceice) [inline lib/util/cache/global/index.ts:11]: Simply test for `global.renovateCache` here, so you are able to override it in the app

Comment (rarkins) [reply lib/util/cache/global/index.ts:11]: True, that's a better stepping stone. I plan to add Redis support here after this

Comment (viceice) [reply lib/util/cache/global/index.ts:11]: What about initialize `renovateCache` in this file init function and simply set `global.renovateCache = fileCache` ?

**Review by viceice — APPROVED** [review]

**Pull request merged**